posthog/models/filters/mixins/common.py

```diff
@@ -554,3 +554,19 @@ class EmailMixin(BaseParamMixin):
     def email(self) -> Optional[str]:
         email = self._data.get("email", None)
         return email
+
+
+class SampleMixin(BaseParamMixin):
+    """
+    Sample factor for a query.
+    """
+
+    default_sample_factor = 0.1
+
+    @cached_property
+    def sample_factor(self) -> Optional[float]:
+        factor = None
+        if self._data.get("sample_results", False):
```

Comment:
Q: Why not send the sample factor from the FE?

Reply:
I'm still unsure what the approach should be with sampling. Should we let the client decide or should we make decisions for them?

@mariusandra seems to like the idea of the client deciding too. From my side, I think I still have non-technical users in mind (i.e. building for PMs), but maybe I should drop that and think about devs first instead. If that's the case, then yeah I'd send this from the frontend and maybe make it a slider-like thing like Marius mentioned

Reply:
Note we can hard-code the value from the frontend if we want and keep api flexibility up :)